.github/workflows/repo-metrics-weekly.yml

```diff
@@ -0,0 +1,58 @@
+name: Weekly repo metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * MON'
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  build:
+    # this workflow will always fail in forks; bail if this isn't running in the upstream
+    if: github.repository == 'aws/aws-cdk'
+    name: metrics
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Get dates for last week
+      shell: bash
+      run: |
+        # Calculate the previous Monday's date
+        PREVIOUS_MONDAY=$(date -d "last monday" "+%Y-%m-%d")
+
+        # Calculate the current Sunday's date
+        CURRENT_SUNDAY=$(date -d "next Sunday" "+%Y-%m-%d")
```

Comment:
QQ: Just would like to know if "next sunday" provides the current sunday or the upcoming sunday? Since the script runs every monday, does this gives the current sunday? We could also get the current sunday may be using "yesterday"?

Reply:
Good point as this would be run every monday, I'll update the naming.